src/app.ts

```diff
@@ -53,45 +56,63 @@ const extractPullRequestInformation = (payload: PullRequestEvent) => {
 const validateChanges = async (
   log: Logger,
   octokit: Pick<OctokitInstance, 'repos'>,
-  configurationChanges: TemplateConfig,
+  configurationChangesOrError: E.Either<YAMLParseError, TemplateConfig>,
 ): Promise<ValidationError[]> => {
   const { getTemplateInformation } = templates(log, octokit)
   const { validateFiles, validateTemplateConfiguration, generateSchema, mergeSchemaToDefault, getDefaultSchema } =
     schemaValidator(log)
 
   const { combineConfigurations } = configuration(log, octokit)
 
-  try {
-    const { configuration: defaultValuesConfiguration, files } = await getTemplateInformation(
-      configurationChanges.repositoryConfiguration.version,
-    )
-
-    const defaultValueSchema = generateSchema(defaultValuesConfiguration.values)
-
-    const combined = combineConfigurations(defaultValuesConfiguration, configurationChanges.repositoryConfiguration)
-
-    const validatedTemplates = validateTemplateConfiguration(
-      {
-        repositoryConfiguration: combined,
-        cstYamlRepresentation: configurationChanges.cstYamlRepresentation,
-      },
-      mergeSchemaToDefault(defaultValueSchema),
-    )
-    const validatedFiles = validateFiles(combined, files)
-
-    return validatedTemplates.errors.concat(validatedFiles.errors)
-  } catch (error) {
-    if (error instanceof VersionNotFoundError) {
-      log.debug('Version for %s/%s:%s has not been found', error.owner, error.name, error.version)
-      const validatedTemplate = validateTemplateConfiguration(configurationChanges, getDefaultSchema())
-      return validatedTemplate.errors.concat({
-        message: `Version could not be found at ${error.owner}/${error.repo}:${error.version}`,
-        line: undefined,
-      })
-    } else {
-      throw error
+  async function validateCorrectYamlChanges(configurationChanges: TemplateConfig) {
+    try {
+      const { configuration: defaultValuesConfiguration, files } = await getTemplateInformation(
+        configurationChanges.repositoryConfiguration.version,
+      )
+
+      const defaultValueSchema = generateSchema(defaultValuesConfiguration.values)
+
+      const combined = combineConfigurations(defaultValuesConfiguration, configurationChanges.repositoryConfiguration)
+
+      const validatedTemplates = validateTemplateConfiguration(
+        {
+          repositoryConfiguration: combined,
+          cstYamlRepresentation: configurationChanges.cstYamlRepresentation,
+        },
+        mergeSchemaToDefault(defaultValueSchema),
+      )
+      const validatedFiles = validateFiles(combined, files)
+
+      return validatedTemplates.errors.concat(validatedFiles.errors)
```

Comment:
Shouldn't this be 

```suggestion
      return E.right(validatedTemplates.errors.concat(validatedFiles.errors))
```

in order to match the return type? 🤔  Or am I completely off? 😄

Reply:
This returns an array of validation errors. We can evaluate to turn everything to `Either` but I feel it falls out of the scope of this PR. WDYT?